R/tidyeval.R

```diff
@@ -231,7 +231,11 @@ fun_name <- function(fun) {
 
 replace_sym <- function(call, sym, replace) {
   if (is_symbol(call, sym)) {
-    replace
+    if (is_list(replace)) {
+      replace[[which(call == syms(sym))]]
```

Comment:
I'm mildly surprised that this equality works because it's comparing a symbol to a list of symbols?

Reply:
Yes, it is comparing a symbol to a list of symbols and it is indeed a bit surprising. But it works nicely:

```r
sym("b") == syms(c("a", "b", "c"))
#> FALSE  TRUE FALSE
```

To make this less surprising it probably makes sense to just use

```r
replace[[match(as_string(call), sym)]]
```

Reply:
Yeah, I'd prefer that.